rust/azure_iot_operations_connector/src/base_connector/managed_azure_device_registry.rs

```diff
@@ -213,39 +218,35 @@ where
 }
 
 /// Azure Device Registry Device Endpoint that includes additional functionality to report status
-pub struct DeviceEndpointClient<T: DataTransformer> {
-    /// The 'name' Field.
-    pub device_name: String,
-    /// The 'endpointName' Field.
-    pub inbound_endpoint_name: String, // needed for easy status reporting?
+#[derive(Debug, Getters)]
+pub struct DeviceEndpointClient {
+    /// The names of the Device and Inbound Endpoint
+    device_endpoint_ref: DeviceEndpointRef,
     /// The 'specification' Field.
-    pub specification: DeviceSpecification,
+    specification: Arc<DeviceSpecification>,
     /// The 'status' Field.
-    pub status: Option<DeviceEndpointStatus>,
-    connector_context: Arc<ConnectorContext<T>>,
+    status: Option<DeviceEndpointStatus>,
```

Comment:
Why is this different than the asset status? For reference: `status: Arc<RwLock<Option<AssetStatus>>>,`

Reply:
Point in time - since the status is only used within the DeviceEndpoint (which isn't clone), it doesn't need to be as locked down (compared to the asset status, which can be updated from the asset or the dataset). 
That being said, I'll go ahead and update this now for symmetry and to enable supporting Clone.